### PR TITLE
Improve nightly CI

### PR DESCRIPTION
**Pull request by mwestphal**
- fix macos arm64 CI #1240 
 - run nightly only on f3d-app/f3d #1236

### CODE REVIEW

**Comment by mwestphal** [comment]
@kidharb FYI

File: .github/workflows/nightly_vtk_master.yml [hunk 1]
```diff
@@ -132,15 +133,44 @@ jobs:
...
+      with:
+        build_type: ${{matrix.build_type}}
+        vtk_version: ${{needs.check_nightly.outputs.vtk_sha}}
+        cpu: ${{matrix.cpu}}
+        lfs_sha: ${{ needs.cache_lfs.outputs.lfs_sha}}
+
+#----------------------------------------------------------------------------
+# MacOS arm64 CI: Build and test
+#----------------------------------------------------------------------------
+  macos:
```
Comment (Meakk) [inline .github/workflows/nightly_vtk_master.yml:167]: Both jobs have the same name?

File: .github/workflows/nightly_vtk_master.yml [hunk 2]
```diff
@@ -132,15 +133,44 @@ jobs:
...
+#----------------------------------------------------------------------------
+  macos_arm:
+    needs: [cache_lfs, check_nightly]
+    if: ${{ needs.check_nightly.outputs.should_run == 'true' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cpu: [arm64]
 
     runs-on: macos-latest
```
Comment (Meakk) [inline .github/workflows/nightly_vtk_master.yml:175]: Should it run on `macos-14`?

```suggestion
    runs-on: macos-14
```

File: .github/workflows/nightly_vtk_master.yml [hunk 3]
```diff
@@ -132,15 +133,44 @@ jobs:
...
+# MacOS CI: Build and test
 #----------------------------------------------------------------------------
   macos:
     needs: [cache_lfs, check_nightly]
     if: ${{ needs.check_nightly.outputs.should_run == 'true' }}
     strategy:
       fail-fast: false
       matrix:
-        cpu: [x86_64, arm64]
+        cpu: [x86_64]
```
Comment (Meakk) [inline .github/workflows/nightly_vtk_master.yml:144]: should we remove the matrix field and just set the cpu directly in `with` on both jobs?

File: .github/workflows/nightly_vtk_master.yml [hunk 4]
```diff
@@ -155,9 +156,36 @@ jobs:
     - name: Generic CI
       uses: ./source/.github/actions/generic-ci
       with:
-        build_type: ${{matrix.build_type}}
         vtk_version: ${{needs.check_nightly.outputs.vtk_sha}}
-        cpu: ${{matrix.cpu}}
+        cpu: ${{inputs.cpu}}
```
Comment (Meakk) [inline .github/workflows/nightly_vtk_master.yml:160]: I meant putting `x86_64` here instead of the matrix

**Review by michaelmigliore — APPROVED** [review]

**Pull request merged**